connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java

```diff
@@ -82,6 +82,12 @@ public void tearDown() {
         return workerProps;
     }
 
+    private Map<String, String> workerPropsWithDynamicPort() {
+        Map<String, String> workerProps = baseWorkerProps();
+        workerProps.put(WorkerConfig.LISTENERS_CONFIG, "HTTP://localhost:0");
```

Comment:
Any reason not to include this in `baseWorkerProps`? That would make it less likely that we regress when people add new tests in this class that bind the port.

Reply:
Only because there are a few test cases which are verifying configurations with/without listeners. It seemed a little awkward to need to remove the property. I could be persuaded though. Maybe these cases should be the exceptions..

Reply:
I see what you mean. I think I'd still do it since the cost of these flaky tests are high and I bet people will use the wrong one if it's available.

Reply:
Fair enough.